File: search/jump_search/java/JumpSearch.java
public class JumpSearch
{
    public static int jumpSearch(int[] arr, int x)
    {
        int n = arr.length;
        // Finding block size to be jumped
        int step = (int)Math.floor(Math.sqrt(n));
        int prev = 0;
        while (arr[Math.min(step, n)-1] < x)
        {
            prev = step;
            step += (int)Math.floor(Math.sqrt(n));
            if (prev >= n)
                return -1;
        }
        // Doing a linear search for x in block beginning with prev.
        while (arr[prev] < x)
        {
            prev++;
            if (prev == Math.min(step, n))
                return -1;
        }
        if (arr[prev] == x)
            return prev; 
        return -1;
    }
 
    // Driver program to test function
    public static void main(String [ ] args)
    {
        int arr[] = { 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610};
        int x = 55;
        int index = jumpSearch(arr, x);
        System.out.println("\nNumber " + x +" is at index " + index);
    }
}

File: search/jump_search/javascript/jumpSearch.js
function jumpSearch(array, needle) {
    var n = array.length;
    var step = Math.floor(Math.sqrt(n));
    var prev = 0;

    while (array[Math.min(step, n) - 1] < needle) {
        prev = step;
        step += Math.floor(Math.sqrt(n));

        if (prev >= n) {
            return false;
        }
    }

    while (array[prev] < needle) {
        prev++;

        if (prev == Math.min(step, n)) {
            return false
        }
    }

    if (array[prev] == needle) {
        return prev;
    }

    return false;
}

var arr = [1, 4, 6, 10, 12, 16, 21, 22, 38, 41, 44, 45];
var search = 6;

var index = jumpSearch(arr, search);

if (index === false) {
    console.log('Number not found');
} else {
    console.log('Number found at index ' + index);
}